ddclient

```diff
@@ -1475,8 +1475,11 @@ sub test_possible_ip {
 
     {
         local $opt{'use'} = 'if';
-        foreach my $if (grep {/^[a-zA-Z]/} `ifconfig -a`) {
-            $if =~ s/:?\s.*//is;
+        my @ifs = map({ /^[^\s:]*:\s*([^\s:]+):/ ? $1 : () } `ip -o link show 2>/dev/null`);
+        @ifs = map({ /^([a-zA-Z].*?)(?::?\s.*)?$/ ? $1 : () } `ifconfig -a`) if $? || !@ifs;
```

Comment:
you need to add `2>/dev/null` to suppress any error message from ifconfig.  I deliberately tested for failure here and got a double warning, one from ifconfig stderr and the one in your code.

Reply:
I intentionally didn't redirect STDERR to /dev/null. In fact, I just changed the `if` command to remove the redirection.

STDERR is important for troubleshooting. Redirecting it to /dev/null is a code smell—there's usually a way to suppress expected errors without also suppressing unexpected errors.

Reply:
I think it is better the way you did it, with `command -v`
Thanks
David